### PR TITLE
Plastic Explosives Update

### PR DESCRIPTION
**Pull request by DaveTheHeadcrab**
Port of /tg/'s plastic explosives, all plastique is now a subtype of grenade to allow for easier maintenance.
Removes the wire datum from plastic explosives, you can now attach assemblies to it instead. Use a wirecutter to remove these assemblies. If you plant plastic explosives that have an assembly attached the brick will not go off until the assembly is triggered. So, for instance, you could set several blocks of C-4 along the walls with remote signaller igniter assemblies attached and trigger them all simultaneously.

Adds a new type of plastic explosives: The X-4 shaped charge. This explosive will breach through a wall and create a sizable explosion, but will not damage anyone on the same side as the person who planted it. This is available to nukeops for 2TC a brick, and death squaddies have had their C-4 replaced with this.

X-4 in action:

*Edit:*
Was bored, made another plastic explosives type as a proof of concept.
C-4 that flashbangs the occupants of the room being breached as it goes off.
Not currently used for anything, but I guess ERT could get it at a later date, maybe.

*Edit 2:*
Oh god I can't stop. Made a new subtype that's got the blast power of C4, but breaches safely like X4.
The flash C4 is now a subtype of that.
Send help.

:cl: DaveTheHeadcrab
tweak: Plastic explosives are now a subtype of grenades. Shouldn't be any noticable difference for players.
tweak: Plastic explosives no longer use the wire datum. You may directly attach an assembly holder (Such as a remote signaller + igniter assembly)
rscadd: Adds X-4 shaped charges, designed for breaching explosively without harming the user. Pick some up at your local nukies r us.
/:cl:

### CODE REVIEW

**Comment by Earthdivine** [comment]
Should bug a spriter to replace the X-4 with those Breach charges from Rainbow Six Seige. Would make it easier to tell rather then Black C4

breach-charge-image

File: code/game/objects/items/weapons/explosives.dm [hunk 1]
```diff
@@ -1,31 +1,96 @@
...
+
+		target.overlays += image_overlay
+		if(!nadeassembly)
+			user << "<span class='notice'>You plant the bomb. Timer counting down from [det_time].</span>"
+			addtimer(src, "prime", det_time*10)
+
+/obj/item/weapon/grenade/plastic/suicide_act(mob/user)
+	message_admins("[key_name_admin(user)](<A HREF='?_src_=holder;adminmoreinfo=\ref[user]'>?</A>) (<A HREF='?_src_=holder;adminplayerobservefollow=\ref[user]'>FLW</A>) suicided with [src.name] at ([user.x],[user.y],[user.z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",0,1)
+	message_admins("[key_name(user)] suicided with [src.name] at ([user.x],[user.y],[user.z])")
+	user.visible_message("<span class='suicide'>[user] activates the [src.name] and holds it above \his head! It looks like \he's going out with a bang!</span>")
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/explosives.dm]: src. reeeeee replace it with `[src]`

File: code/game/objects/items/weapons/explosives.dm [hunk 2]
```diff
@@ -1,31 +1,96 @@
...
+		log_game("[key_name(user)] planted [src.name] on [target.name] at ([target.x],[target.y],[target.z]) with [det_time] second fuse")
+
+		target.overlays += image_overlay
+		if(!nadeassembly)
+			user << "<span class='notice'>You plant the bomb. Timer counting down from [det_time].</span>"
+			addtimer(src, "prime", det_time*10)
+
+/obj/item/weapon/grenade/plastic/suicide_act(mob/user)
+	message_admins("[key_name_admin(user)](<A HREF='?_src_=holder;adminmoreinfo=\ref[user]'>?</A>) (<A HREF='?_src_=holder;adminplayerobservefollow=\ref[user]'>FLW</A>) suicided with [src.name] at ([user.x],[user.y],[user.z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",0,1)
+	message_admins("[key_name(user)] suicided with [src.name] at ([user.x],[user.y],[user.z])")
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/explosives.dm]: same hereeeeeeeeeeeeeeee

File: code/game/objects/items/weapons/explosives.dm [hunk 3]
```diff
@@ -1,31 +1,96 @@
...
+		message_admins("[key_name_admin(user)](<A HREF='?_src_=holder;adminmoreinfo=\ref[user]'>?</A>) (<A HREF='?_src_=holder;adminplayerobservefollow=\ref[user]'>FLW</A>) planted [src.name] on [target.name] at ([target.x],[target.y],[target.z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[target.x];Y=[target.y];Z=[target.z]'>JMP</a>) with [det_time] second fuse",0,1)
+		log_game("[key_name(user)] planted [src.name] on [target.name] at ([target.x],[target.y],[target.z]) with [det_time] second fuse")
+
+		target.overlays += image_overlay
+		if(!nadeassembly)
+			user << "<span class='notice'>You plant the bomb. Timer counting down from [det_time].</span>"
+			addtimer(src, "prime", det_time*10)
+
+/obj/item/weapon/grenade/plastic/suicide_act(mob/user)
+	message_admins("[key_name_admin(user)](<A HREF='?_src_=holder;adminmoreinfo=\ref[user]'>?</A>) (<A HREF='?_src_=holder;adminplayerobservefollow=\ref[user]'>FLW</A>) suicided with [src.name] at ([user.x],[user.y],[user.z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",0,1)
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/explosives.dm]: THIS ONE TOO DON'T THINK YOU CAN RUN

File: code/game/objects/items/weapons/explosives.dm (hunk 1 above)
Comment (tigercat2000) [inline code/game/objects/items/weapons/explosives.dm]: wait oh god this is fucking to use the \he aaaaa

File: code/game/objects/items/weapons/explosives.dm (hunk 3 above)
Comment (Spacemanspark) [inline code/game/objects/items/weapons/explosives.dm]: I'm running.

**Comment by Chakirski** [comment]
Did someone say sprites?

**Comment by Fox-McCloud** [comment]
Current sprites are fine.

They're pitch black with a flashing red+yellow light vs green and white. If someone can't make the distinction between those...

**Comment by Earthdivine** [comment]
But it would be so much cooler.

**Comment by Necaladun** [comment]
I approve of explosives.

**Pull request merged**